q2cli/core/config.py

```diff
@@ -0,0 +1,100 @@
+                    'bright_magenta', 'bright_cyan', 'bright_white']
+    valid_booleans = {'true': True,
+                      'false': False,
+                      't': True,
+                      'f': False}
+    styles = {'options': {'fg': 'blue'},
+              'type': {'fg': 'green'},
+              'default_args': {'fg': 'magenta'},
+              'command': {'fg': 'blue'},
+              'emphasis': {'underline': True},
+              'problem': {'fg': 'yellow'},
+              'errors': {'fg': 'red', 'bold': True},
+              'required': {'underline': True},
+              'success': {'fg': 'green'}}
+
+    def __init__(self):
+        self.parse_file()
+
+    def build_error(self, current, valid_list, valid_string):
+        valids = ''
+        for valid in valid_list:
+            valids += valid + '\n'
+        raise ValueError(f'{current!r} is not a {valid_string}. The '
+                         f'{valid_string}s are:\n{valids}')
+
+    def parse_file(self, fp=path):
+        try:
+            if os.path.exists(fp):
+                parser = configparser.ConfigParser()
+                parser.read(fp)
+                for type_user in parser.sections():
+                    type = type_user.lower()
+                    if type not in self.valid_types:
+                        self.build_error(type_user, self.valid_types,
+                                         'valid type')
+                    for styling_user in parser[type]:
+                        styling = styling_user.lower()
+                        if styling not in self.valid_stylings:
+                            self.build_error(styling_user, self.valid_stylings,
+                                             'valid styling')
+                        val_user = parser[type][styling]
+                        val = val_user.lower()
+                        if styling == 'fg' or styling == 'bg':
+                            if val not in self.valid_colors:
+                                self.build_error(val_user, self.valid_colors,
+                                                 'valid color')
+                        else:
+                            if val not in self.valid_booleans:
+                                self.build_error(val_user, self.valid_booleans,
+                                                 'valid boolean')
+                            val = self.valid_booleans[val]
+                        self.styles[type][styling] = val
+            else:
+                # If the bad path is the default path they don't have a custom
+                # config. If the bad path isn't the default path they tried to
+                # import a nonexistent file and we want to error
+                if fp != self.path:
+                    raise ValueError(f'{fp!r} is not a valid filepath.')
+        except Exception as e:
+            if os.path.exists(fp):
+                os.unlink(fp)
```

Comment:
This was a hold-over from a previous way I was doing things. At first, I was importing the file to the config path then parsing it, so if it was a bad file I wanted it removed from that location or your command line wouldn't work anymore because every time you tried to run a command it would attempt to parse the bad file and crash. Now I just make sure the file is parsable before moving it into that directory, so unlinking the file there is no longer needed. It has been changed.